sdk/core/Azure.Core/tests/ModelSerializationTests/Animal.cs

```diff
@@ -159,6 +159,43 @@ public bool TrySerialize(Stream stream, out long bytesWritten, SerializableOptio
                 return false;
             }
         }
+
+        public void Deserialize(Stream stream, SerializableOptions options = default)
+        {
+            try
+            {
+                JsonDocument jsonDocument = JsonDocument.Parse(stream);
```

Comment:
we should change the try implementation to call this instead of duplicating the implementation.